src/index.ts

```diff
@@ -1,95 +1,18 @@
-            }
-        })
-        .file("file-from-env", { file: process.env.CUSTOM_CONFIG_LOCATION ?? "config/custom.json" })
-        .file("config-env-file", { file: `config/${process.env.NODE_CONFIG_ENV}.json` })
-        .file("default-file", { file: "config/default.json" });
-
-    const connectorConfig = nconf.get() as ConnectorRuntimeConfig;
-    addCorrelationIdSupportToLogger(connectorConfig);
-
-    if (connectorConfig.modules.sync.enabled && connectorConfig.modules.sse.enabled) {
-        // eslint-disable-next-line no-console
-        console.warn("The SSE and Sync modules cannot be enabled at the same time, the Sync module will be disabled.");
-        connectorConfig.modules.sync.enabled = false;
-    }
-
-    return connectorConfig;
-}
-
-function addCorrelationIdSupportToLogger(connectorConfig: ConnectorRuntimeConfig) {
-    Object.entries(connectorConfig.logging.appenders).forEach(([_key, appender]) => {
-        if ("layout" in appender && appender.layout.type === "pattern") {
-            const tokens = appender.layout.tokens;
-
-            appender.layout.tokens = {
-                ...tokens,
-                correlationId: () => {
-                    return correlator.getId() ?? "";
-                }
-            };
-        }
-    });
-}
-
-const envKeyMapping: Record<string, string> = {
-    DATABASE_NAME: "database:dbName", // eslint-disable-line @typescript-eslint/naming-convention
-    API_KEY: "infrastructure:httpServer:apiKey", // eslint-disable-line @typescript-eslint/naming-convention
-    DATABASE_CONNECTION_STRING: "database:connectionString", // eslint-disable-line @typescript-eslint/naming-convention
-    SYNC_ENABLED: "modules:sync:enabled", // eslint-disable-line @typescript-eslint/naming-convention
-    PLATFORM_CLIENT_ID: "transportLibrary:platformClientId", // eslint-disable-line @typescript-eslint/naming-convention
-    PLATFORM_CLIENT_SECRET: "transportLibrary:platformClientSecret", // eslint-disable-line @typescript-eslint/naming-convention
-    DEBUG: "debug" // eslint-disable-line @typescript-eslint/naming-convention
-};
-
-function applyAlias(variable: { key: string; value: any }) {
-    if (envKeyMapping[variable.key]) {
-        variable.key = envKeyMapping[variable.key];
-    }
-}
-
-function parseString(value: string) {
-    try {
-        return JSON.parse(value);
-    } catch (_) {
-        return value;
-    }
-}
-
-async function run() {
-    const config = createConnectorConfig();
-    if (!config.debug) process.env.TLS_REJECT_UNAUTHORIZED = "1";
-
-    const runtime = await ConnectorRuntime.create(config);
-    await runtime.start();
-}
-
-run()
-    .then()
+#!/usr/bin/env node
+
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+import { startConnectorCommand } from "./cli/commands";
+
+yargs(hideBin(process.argv))
+    .command(startConnectorCommand)
+    .demandCommand(1, 1, "Please specify a command")
+    .scriptName("nmshd-connector")
```

Comment:
I have one last thing, when using the image to run the CLI `nmshd-connector` is not fitting.

If you change this to 
```suggestion
    .scriptName("")
```

the connector outputs

```
> docker run --rm ghcr.io/nmshd/connector:prod --help   
 <command>

Commands:
  start  start the connector

Options:
      --version  Show version number                                   [boolean]
  -h, --help     Show help                                             [boolean]
``` 

and completely omits the command name and therefore doesn't confuses the user.